### DISCUSSION TITLE
GeneActivity on pseudo-bulk data

### DISCUSSION THREAD

**Discussion by jgarces02** [stuart-lab/signac#1750] [Q&A]
Hi there, 

I calculated the gene activity scores for my dataset and realized there was a bias toward more abundant populations. Is there any way of performing this calculation on the averaged accessibility? Something like `Seurat`'s `AggregateExpression`?

Thanks in advance!

**Comment by jgarces02** [comment]
I tried this, does it make sense?

```
gene_activities <- GeneActivity(seuratobj_atac, biotypes = NULL, max.width = NULL)

seuratobj_atac[["gene_activities"]] <- CreateAssayObject(counts = gene_activities)
DefaultAssay(seuratobj_atac) <- "gene_activities"

seuratobj_atac_pseudobulk <- AggregateExpression(seuratobj_atac, assays = "gene_activities", return.seurat = T, 
	group.by = c("cell_type", "patient", "timepoint"))

```

**Reply by timoast** [reply]
I think `AggregateExpression` will sum for each group rather than return the mean. You might want to check `AverageExpression` instead
